ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision/BulkSync.hs

```diff
@@ -0,0 +1,605 @@
+--   have been requested and are expected to arrive but have not yet.
+--
+-- - Let @peersOrder@ be an order of preference among the peers. This order is
+--   not set in stone and will evolve as we go.
+--
+-- - Let @currentPeer :: Maybe peer@ be the “current peer” with which we are
+--   interacting. If it exists, this peer must be the best according to
+--   @peersOrder@, and the last fetch request must have been sent to them.
+--
+-- - Let @currentStart :: Time@ be the latest time a fetch request was sent
+--   while there were no outstanding blocks.
+--
+-- - Let @gracePeriod@ be a small duration (eg. 10s), during which a “cold” peer
+--   is allowed to warm up (eg. grow the TCP window) before being expected to
+--   feed blocks faster than we can validate them.
+--
+-- One iteration of this decision logic:
+--
+-- 0. If @inflight(currentPeer)@ is non-empty and the block validation component
+--    has idled at any point after @currentStart@ plus @gracePeriod@, then the
+--    peer @currentPeer@ has failed to promptly serve @inflight(currentPeer)@,
+--    and:
+--
+--   - If @currentPeer@ is the ChainSync Jumping dynamo, then it must
+--     immediately be replaced as the dynamo.
+--
+--   - Stop considering the peer “current” and make them the worst according to
+--     the @peersOrder@.
+--
+-- 1. Select @theCandidate :: AnchoredFragment (Header blk)@. This is the best
+--    candidate header chain among the ChainSync clients (eg. best raw
+--    tiebreaker among the longest).
+--
+-- 2. Select @thePeer :: peer@. If @inflight(currentPeer)@ is not empty, then
+--    this is @currentPeer@. Otherwise:
+--
+--    - Let @grossRequest@ be the oldest blocks on @theCandidate@ that have not
+--      already been downloaded and total less than 20 mebibytes.
+--
+--    - If @grossRequest@ is empty, then terminate this iteration. Otherwise,
+--      pick the best peer (according to @peersOrder@) offering all of the
+--      blocks in @grossRequest@.
+--
+-- 3. Craft that actual request to @thePeer@ asking blocks of @theCandidate@:
+--
+--    - If the byte size of @inflight(thePeer)@ is below the low-water mark,
+--      then terminate this iteration.
+--
+--    - Decide and send the actual next batch request, as influenced by exactly
+--      which blocks are actually already currently in-flight with @thePeer@.
+--
+-- 4. If we went through the election of a new peer, replace @currentPeer@ and
+--    reset @currentStart@. REVIEW: Maybe this should just be done directly in
+--    step 2.
+--
+-- Terminate this iteration.
+--
+-- About ignored in-flight requests
+-- --------------------------------
+--
+-- One can note that in-flight requests are ignored when finding a new peer, but
+-- considered when crafting the actual request to a chosen peer. This is by
+-- design. The goal of this algorithm is to keep talking to the same peer unless
+-- it proves to be too weak; in that case, @inflight(p)@ will be empty for all
+-- @p /= currentPeer@.
```

Comment:
```suggestion
-- About the influence of in-flight requests
-- -----------------------------------------
--
-- In-flight requests are ignored when finding a new peer, but
-- considered when crafting the actual request to a chosen peer. We explain the
-- rationale here.
```

I couldn't make sense last sentence of the paragraph. It looks to me like non-current peers can have in-flights regardless of what the current peer is doing. It also looks like the narrative stands without that consideration.